### PR TITLE
[tests-only] [full-ci]tests for occ command `files_external`

### PR DESCRIPTION
**Pull request by sakshamgurung**
## Description

Acceptance tests for occ command `files_external` and its variations.
## Related Issue

- PR for https://github.com/owncloud/core/issues/39195

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature [hunk 1]
```diff
@@ -61,3 +61,49 @@ Feature: using files external service with storage as webdav_owncloud
...
+        | authentication_backend | password::password |
+      And user "admin" has uploaded file with content "Hello from Local!" to "TestMountPoint/test.txt"
+      And user "Brian" has been created with default attributes and without skeleton files
+      And the administrator has added user "Brian" as the applicable user for local storage mount "TestMountPoint"
+      And the following users have been listed as applicable for local storage mount "TestMountPoint":
+        | users  | Brian |
+      When the administrator removes user "Brian" as the applicable user for local storage mount "TestMountPoint" using the occ command
+      Then the command should have been successful
+      And the applicable users list should be empty for local storage mount "TestMountPoint"
+      And as "Brian" file "TestMountPoint/test.txt" should not exist
```
Comment (sakshamgurung) [inline tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature:107]: @kiranparajuli589 Why Brian can access file even after removing from accessible user's list.

**Comment by sakshamgurung** [comment]
I converted to draft because we do not want to merge acceptance test refactorings to master while the 10.9.0 release process is in progress.

File: tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature [hunk 2]
```diff
@@ -61,3 +61,198 @@ Feature: using files external service with storage as webdav_owncloud
...
+      | mount_point            | TestMountPoint     |
+      | authentication_backend | password::password |
+    And user "admin" has uploaded file with content "Hello from Local!" to "TestMountPoint/test.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    And the administrator has added group "grp1" as the applicable group for local storage mount "TestMountPoint"
+    And the administrator has added user "admin" as the applicable user for local storage mount "TestMountPoint"
+    And the following groups have been listed as applicable for local storage mount "TestMountPoint":
+      | groups  | grp1 |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature:147]: Nice, you've to use verifications for the Given steps, but please write use this verification inside the given step's implementation, not as a different step.

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 3]
```diff
@@ -1559,6 +1619,82 @@ public function theAdminAddsRemovesAsTheApplicableUserForMountUsingTheOccCommand
...
+		$actualApplicable = $this->getListOfApplicableUserOrGroupForMount($localStorage);
+		if (!isset($actualApplicable)) {
+			return;
+		} else {
+			Assert::assertFalse(
+				\array_key_exists($usersOrGroups, $actualApplicable),
+				"Command outputs the following applicable users and groups: "
+				. \json_decode($this->featureContext->getStdOutOfOccCommand())
+			);
+		}
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/OccContext.php:1695]:
```suggestion
		if (isset($actualApplicable)) {
			Assert::assertFalse(
				\array_key_exists($usersOrGroups, $actualApplicable),
				"Command outputs the following applicable users and groups: "
				. \json_decode($this->featureContext->getStdOutOfOccCommand())
			);
		}
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -1559,6 +1615,82 @@ public function theAdminAddsRemovesAsTheApplicableUserForMountUsingTheOccCommand
...
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theApplicableUsersOrGroupsListShouldBeEmptyForLocalStorageMount(string $usersOrGroups, string $localStorage): void {
+		$actualApplicable = $this->getListOfApplicableUserOrGroupForMount($localStorage);
+		if (isset($actualApplicable)) {
+			Assert::assertFalse(
+				\array_key_exists($usersOrGroups, $actualApplicable),
+				"Command outputs the following applicable users and groups: "
+				. \json_decode($this->featureContext->getStdOutOfOccCommand())
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/OccContext.php:1689]:
```suggestion
				"Expected the application users/groups list to be empty, but found: \n "
				. \json_decode($this->featureContext->getStdOutOfOccCommand())
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 5]
```diff
@@ -1603,6 +1735,33 @@ public function theAdminHasAddedRemovedTheApplicableUserForMountUsingTheOccComma
...
+					. " is not present in the actual list of applicable "
+					. $userOrGroup
+					. " for mount point: "
+					. $mount . ".\n"
+				);
+			} else {
+				throw new Exception(
+					"Got actual list of applicable "
+					. $userOrGroup
+					. " null"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/OccContext.php:1761]:
```suggestion
					"Expected applicable list to contain " . $userOrGroup . " but found empty applicable list"
```

**Comment by sakshamgurung** [comment]
Some random fails. Please restart CI

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/33940/75/7
`Error response from daemon: No such image: owncloudci/nodejs:14`
"random"  drone fail. I restarted CI.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/33993/121/18
The new tests ran and passed.
Now to review...

**Comment by kiranparajuli589** [comment]
🔄 @phil-davis I updated the code with some changes _(see latest commits)_. Please review again.

**Review by phil-davis — APPROVED** [review]
LGTM. I will set to draft and cherry-pick to branch `acceptance-test-changes-waiting-2021-11`

**Comment by phil-davis** [comment]
cherry-picked to #39514 
I am leaving this draft for now. Hopefully that PR gets merged soon. Then we can close all these draft PRs.

**Comment by phil-davis** [comment]
cherry-picked to #39514 
I am leaving this draft for now. Hopefully that PR gets merged soon. Then we can close all these draft PRs.

**Comment by phil-davis** [comment]
cherry-picked to #39514 
I am leaving this draft for now. Hopefully that PR gets merged soon. Then we can close all these draft PRs.

**Comment by phil-davis** [comment]
Merged to master in #39612

**Pull request closed without merging**